src/apps/backend/modules/account/account_service.py

```diff
@@ -23,16 +25,20 @@ def get_account_by_phone_number(*, phone_number: PhoneNumber) -> Account:
         return AccountReader.get_account_by_phone_number(phone_number=phone_number)
 
     @staticmethod
-    def get_or_create_account_by_phone_number(*, params: CreateAccountByPhoneNumberParams) -> Account:
+    def get_or_create_account_by_phone_number(
+        *, params: CreateAccountByPhoneNumberParams
+    ) -> Tuple[Account, Optional[Otp]]:
         account = AccountReader.get_account_by_phone_number_optional(phone_number=params.phone_number)
 
         if account is None:
             account = AccountWriter.create_account_by_phone_number(params=params)
 
         create_otp_params = CreateOtpParams(phone_number=params.phone_number)
-        OtpService.create_otp(params=create_otp_params)
-
-        return account
+        otp = OtpService.create_otp(params=create_otp_params)
+        
+        otp_to_return = otp if os.environ.get("APP_ENV", "development") not in ["production", "preview"] else None
```

Comment:
My goodness, this defeats the purpose of the having a config system.

We need to add configuration in config system and based on the flag do the right thing. The app should almost never have to write code that checks env specifically which kinda defeats the purpose of having a config system in the first place